### PR TITLE
Backup

### PR DESCRIPTION
**Pull request by chris-sanders**
This is built on top of PR #6 which just fixes the functional tests. This adds layer-backup, which provides an action to run backups. This doesn't provide off-machine backups necessarily, but allows you to setup backups to a folder, with the intention being that you backup to a network mount.

### CODE REVIEW

File: templates/gitlab.rb.j2 [hunk 1]
```diff
@@ -322,7 +322,7 @@ redis_exporter['enable'] = false
 # gitlab_rails['backup_pg_schema'] = 'public'
 
 ###! The duration in seconds to keep backups before they are allowed to be deleted
-# gitlab_rails['backup_keep_time'] = 604800
+gitlab_rails['backup_keep_time'] = 1
```
Comment (devec0) [inline templates/gitlab.rb.j2]: I believe setting this to 1 will have the effect of subsequent runs of the GitLab backup command removing all old backups each time. Is this what you intended?

File: tests/functional/test_deploy.py [hunk 2]
```diff
@@ -54,19 +55,21 @@ def source(request):
...
 
+@pytest.mark.timeout(30)
 async def test_redis_deploy(model, series, app, request):
     """Create a Redis deployment for testing relation to GitLab."""
     if app.name.endswith("jujucharms"):
         pytest.skip("No need to test the charm deploy")
 
     application_name = "gitlab-redis-{}".format(series)
-    cmd = "juju deploy cs:~omnivector/redis -m {} --series xenial {}".format(
+    cmd = "juju deploy cs:~redis-charmers/redis -m {} --series xenial {}".format(
```
Comment (devec0) [inline tests/functional/test_deploy.py]: This seems like it should be in #6 - or does the redis change need to be merged before this one?

**Comment by devec0** [comment]
Couple of comments but otherwise looks good

File: tests/functional/test_deploy.py (hunk 2 above)
Comment (chris-sanders) [reply tests/functional/test_deploy.py]: The redis change has to land first, w/o that branch functional tests fail and there is a functional test for the backup action that I'd like to see passing.

File: templates/gitlab.rb.j2 (hunk 1 above)
Comment (chris-sanders) [reply templates/gitlab.rb.j2]: Yup that's the intention. 

Layer-backup will run the gtilab backup and then tar the results with the config files on each run. The final backup location is pruned based on a config value for the number of backups you want to keep. No reason to keep multiple copies of the database backup in each of the final tar files.

Comment (devec0) [reply templates/gitlab.rb.j2]: OK makes sense

**Comment by chris-sanders** [comment]
I noticed the redis branch landed, and the diff didn't update. I changed the base branch away and back to master and now it's updated. If I hadn't included some 'Black' changes it would be even cleaner.

**Pull request merged**